samtranslator/model/stepfunctions/events.py

```diff
@@ -135,8 +137,10 @@ def to_cloudformation(self, resource, **kwargs):  # type: ignore[no-untyped-def]
         events_rule.Name = self.Name
         events_rule.Description = self.Description
 
-        role = self._construct_role(resource, permissions_boundary)  # type: ignore[no-untyped-call]
-        resources.append(role)
+        role = None
+        if self.Role is None:
+            role = self._construct_role(resource, permissions_boundary)  # type: ignore[no-untyped-call]
```

Comment:
Generally we try to avoid adding `# type: ignore` in new code. Have you tried to type `_construct_role()`? `# type: ignore` is the last resort if typing the invoked API would take too long.